eos-tech-support/eos-download-image

```diff
@@ -29,45 +30,72 @@ def check_call(*args):
     subprocess.check_call(args, stdout=sys.stderr)
 
 
-def ensure_keyring():
+def url_dirname(url):
+    raw_path = urllib.parse.urlparse(url).path
+    path = urllib.parse.unquote(raw_path)
+    return os.path.dirname(path.lstrip('/'))
+
+
+def ensure_keyring(destdir=''):
     if os.path.isfile(KEYRING_SYSTEM_PATH):
         return KEYRING_SYSTEM_PATH
 
-    keyring = os.path.abspath(KEYRING_FILENAME)
+    # The absolute path is needed so that gpgv doesn't try to use a
+    # keyring from the gnupg homedir
+    keyring = os.path.abspath(os.path.join(destdir, KEYRING_FILENAME))
     if not os.path.isfile(keyring):
         log("Fetching Endless keyring")
         check_call("wget", "-O", keyring, KEYRING_URL)
 
     return keyring
 
 
-def download_and_verify(base, meta, what='image'):
+def download_and_verify(base, meta, what='image', destdir='',
+                        mirror=False):
     image_url = '{}/{}'.format(base, meta['file'])
     signature_url = '{}/{}'.format(base, meta['signature'])
 
-    image = os.path.basename(image_url)
+    image_dir = signature_dir = ''
+    if mirror:
+        image_dir = url_dirname(image_url)
+        signature_dir = url_dirname(signature_url)
+    image = os.path.join(destdir, image_dir, os.path.basename(image_url))
+    signature = os.path.join(destdir, signature_dir,
+                             os.path.basename(signature_url))
 
     log("Downloading compressed", what, "from", image_url)
-    check_call("wget", "--continue", image_url)
+    full_image_dir = os.path.dirname(image)
+    if full_image_dir:
+        os.makedirs(full_image_dir, exist_ok=True)
+    check_call("wget", "--continue", "-O", image, image_url)
 
     if meta.get('extracted_size'):
         with open(image + '.size', 'w') as f:
             f.write(str(meta['extracted_size']))
 
     log("Downloading compressed", what, "from", signature_url)
```

Comment:
Just noticed this should be `"Downloading compressed", what, "signature from", signature_url`, any chance you could add this fix to this branch?